src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/api/jira/JiraSharedStorageResource.kt

```diff
@@ -0,0 +1,8 @@
+package com.atlassian.performance.tools.awsinfrastructure.api.jira;
+
+/**
+ * Describes a set of resources used by Jira and stored in shared storage (eg. shared home or S3).
+ */
+enum class JiraSharedStorageResource {
```

Comment:
Can this enum change in the future? If so then `enum` is not a great option for it, as changes to it would break API.

Usually polymprphism is a good way to avoid `enum` usage. If you want I could help with figuring out how a polymprphic model could look like here.

We could proceed with `enum` if it's very unlikely for the value set to change.

Reply:
It's definitely possible for this to change.  Is there an example of the polymorphic approach you're suggesting and I could have a go at implementing it?

Reply:
What I was thinking about was a model where a polymorphic replacement for `JiraSharedStorageResource` could be an extension to `JiraHomeSource`. Both operations based on it that you implemented so far are done on Jira home, so I believe that it should logically fit there.

Each Jira resource type (avatars, attachments) could be implemented as wrapper which would delegate to any dataset. It would first let the delegate configure the home on remote and then it would create S3 bucket and upload selected  artifacts from Jira home there. Then it could append (or create) `filestore-config.xml`.

When I think about it now this may seem to be a bit of implementation changes. I'd personally still go for it, though maybe there is a simpler way to avoid `enum`s that the team can think of. Alternatively maybe `enum` are good enough for now - still need to consult with the team (we have a standup in 1h).

Reply:
I was trying to show the idea with this change set, however I wasn't able to finish it: https://github.com/atlassian/aws-infrastructure/pull/168

Reply:
@MatthewCochrane your alternative solutions to getting rid of `enum` usage that you posted in [this comment](https://github.com/atlassian/aws-infrastructure/pull/168#issuecomment-1566580555) should be fine. Please proceed with any solution that will work for you and don't be blocked by my suggestions. The main concern was about `enum` in API and usage of polymorphism is not mandatory.

Reply:
Thanks, have pushed a change.